# Program Hampiran Fungsi Cos

Program ini dibuat dengan menggunakan deret Taylor di sekitar 0 (deret Maclaurin). Bentuk dari hampiran fungsi $\cos{(x)}$ dalam deret Maclaurin dapat dituliskan sebagai berikut:
$$ \cos{(x)}=1-\frac{x^2}{2!}+\frac{x^4}{4!}-\frac{x^6}{6!}+R_n(x)\tag{1}$$

Deret Maclaurin pada dasarnya tak berhingga, namun dengan menetapkan suatu parameter yang berkaitan dengan galat maka perhitungan dari deret tersebut dapat dibatasi sampai suku atau orde tertentu sesuai dengan kriteria yang telah ditetapkan sebelumnya. Parameter tersebut dapat disebut sebagai epsilon yang mewakili nilai suku terakhir dalam deret.

In [1]:
import math
#Parameter
epsilon = 0.00000001 #ini dapat diatur sesuai kebutuhan
f = 0
suku = 1.0
pangkat = 0
tanda = 1.0
count = 1


x = 0.2 #Ini dapat diganti sesuai kebutuhan



while suku >= epsilon:
    f += tanda*suku
    print("Hasil hampiran hingga suku ke -",count, "adalah",f)
    
    tanda *= -1.0
    pangkat += 2
    suku = (x**pangkat)/math.factorial(pangkat)
    
    count += 1
    
print("Hasil akhir hampiran adalah",round(f,7))
     

Hasil hampiran hingga suku ke - 1 adalah 1.0
Hasil hampiran hingga suku ke - 2 adalah 0.98
Hasil hampiran hingga suku ke - 3 adalah 0.9800666666666666
Hasil hampiran hingga suku ke - 4 adalah 0.9800665777777777
Hasil akhir hampiran adalah 0.9800666


Untuk nilai $x=0.2$ dan $epsilon = 0.00000001$ hasil akhir yang diperoleh sebesar $cos(0.2)\approx \boxed{0.9800666}$